chef/cookbooks/ceph/recipes/radosgw_civetweb.rb

```diff
@@ -0,0 +1,79 @@
+# This recipe doesn't do anything other than deal with SSL keys and certificates.
+# Everything else for civetweb setup is in conf.rb.  It could arguably be merged
+# into radosgw.rb; it only exists as a separate file because it used to be
+# radosgw_apache2.rb
+
+if node["ceph"]["radosgw"]["ssl"]["enabled"]
+  certfile      = node["ceph"]["radosgw"]["ssl"]["certfile"]
+  keyfile       = node["ceph"]["radosgw"]["ssl"]["keyfile"]
+  pemfile       = node["ceph"]["radosgw"]["ssl"]["pemfile"]
+  if  node["ceph"]["radosgw"]["ssl"]["generate_certs"]
+    package "openssl"
+    ruby_block "generate_certs for radosgw" do
+        block do
+          unless ::File.exists? certfile and ::File.exists? keyfile
+            require "fileutils"
+
+            Chef::Log.info("Generating SSL certificate for radosgw...")
+
+            [:certfile, :keyfile].each do |k|
+              dir = File.dirname(node[:ceph][:radosgw][:ssl][k])
+              FileUtils.mkdir_p(dir) unless File.exists?(dir)
+            end
+
+            # Generate private key
+            %x(openssl genrsa -out #{keyfile} 4096)
+            if $?.exitstatus != 0
+              message = "SSL private key generation failed"
+              Chef::Log.fatal(message)
+              raise message
+            end
+            FileUtils.chown "root", node[:ceph][:group], keyfile
+            FileUtils.chmod 0640, keyfile
+
+            # Generate certificate signing requests (CSR)
+            conf_dir = File.dirname certfile
+            ssl_csr_file = "#{conf_dir}/signing_key.csr"
+            ssl_subject = "\"/C=US/ST=Unset/L=Unset/O=Unset/CN=#{node[:fqdn]}\""
+            %x(openssl req -new -key #{keyfile} -out #{ssl_csr_file} -subj #{ssl_subject})
+            if $?.exitstatus != 0
+              message = "SSL certificate signed requests generation failed"
+              Chef::Log.fatal(message)
+              raise message
+            end
+
+            # Generate self-signed certificate with above CSR
+            %x(openssl x509 -req -days 3650 -in #{ssl_csr_file} -signkey #{keyfile} -out #{certfile})
```

Comment:
Metrics/LineLength: Line is too long. [101/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)<br>Style/CommandLiteral: Use backticks around command string. (https://github.com/bbatsov/ruby-style-guide#percent-x)